core/src/mindustry/logic/LExecutor.java

```diff
@@ -1727,6 +1727,32 @@ public void run(LExecutor exec){
             }
         }
     }
-
+    
+    public static class ApplyWeatherI implements LInstruction{
+        public String weather;
```

Comment:
This should be a variable that points to a weather constant like every other instruction, not a hard-coded string

Reply:
Isn't this what Apply Effect used?